Helper/PaymentMethods.php

```diff
@@ -165,95 +163,119 @@ protected function fetchAlternativeMethods($country)
             return [];
         }
 
-        $currencyCode = $this->getCurrentCurrencyCode($store);
-
-        $adyFields = [
+        $request = [
             "channel" => "Web",
             "merchantAccount" => $merchantAccount,
             "countryCode" => $this->getCurrentCountryCode($store, $country),
-            "amount" => [
-                "currency" => $currencyCode,
-                "value" => $this->adyenHelper->formatAmount(
-                    $this->getCurrentPaymentAmount(),
-                    $currencyCode
-                ),
-            ],
-            "shopperReference" => $this->getCurrentShopperReference(),
-            "shopperLocale" => $this->adyenHelper->getCurrentLocaleCode($store->getId())
         ];
 
+        if (!empty($this->getCurrentShopperReference())) {
+            $request["shopperReference"] = $this->getCurrentShopperReference();
+        }
+
+        if (!empty($this->adyenHelper->getCurrentLocaleCode($store->getId()))) {
+            $request["shopperLocale"] = $this->adyenHelper->getCurrentLocaleCode($store->getId());
+        }
+        $currencyCode = $this->getCurrentCurrencyCode($store);
```

Comment:
can the getCurrentCurrencyCode return null?

Reply:
No, if it is empty it will return the base currency